src/toil/lib/bioio.py

```diff
@@ -283,6 +283,11 @@ def parseBasicOptions(parser):
 
     setLoggingFromOptions(options)
 
+    # Look for a default zone.
```

Comment:
This works, but it should be fetched via a function that checks both the environment variable and the user options.

In this case, if the `TOIL_AZURE_ZONE` has a default value (now or added in the future) it will overwrite what the user specifies via `--zone uswest`, which may not be the behavior we want since then the user can't specify a zone via the commandline anymore once this is set.

There should be a function that checks all places (options and env variable) and returns the zone.  See here for how we fetch the AWS zone:
https://github.com/BD2KGenomics/toil/blob/9455dd4c2662979da9ff18eef8bf5bf7b5f10344/src/toil/provisioners/aws/__init__.py#L37

There needs to be a function like that for Azure.  This file might be a good place to put it:
https://github.com/BD2KGenomics/toil/blob/9455dd4c2662979da9ff18eef8bf5bf7b5f10344/src/toil/provisioners/azure/__init__.py  (include the toil copyright header there if you do).